com.unity.sg2/Editor/GraphDeltaRegistry/Registry/GraphType.cs

```diff
@@ -233,25 +251,18 @@ public enum Usage { In, Out, Static, Local }
 
         public void BuildType(FieldHandler field, Registry registry)
         {
-            // default initialize to a float4;
+            // TODO: Default initialization should be a non-dynamic scalar single precision float.
             field.AddSubField(kPrecision, Precision.Single, true);
             field.AddSubField(kPrimitive, Primitive.Float, true);
+            // TODO: Length should start at 1.
             field.AddSubField(kLength, Length.Four, true);
             field.AddSubField(kHeight, Height.One, true);
 
+            field.AddSubField(kDynamicLength, false, true);
+            field.AddSubField(kDynamicHeight, false, true);
```

Comment:
Added specific flags for dynamic dimensions. The `Any` value doesn't cover enough dynamic cases when it comes to accepting new candidate edges (eg. if an any field of a port is concretized, we don't know that it is dynamic anymore and can't allow connections that would normally be fine).